meteor/client/ui/SegmentTimeline/SegmentTimelineContainer.tsx

```diff
@@ -794,11 +816,14 @@ export const SegmentTimelineContainer = translateWithTracker<IProps, IState, ITr
 						? partOffset + e.detail.currentTime - virtualStartedPlayback + lastTakeOffset
 						: partOffset + lastTakeOffset
 
+				const budgetDuration = this.getSegmentBudgetDuration()
```

Comment:
Why do this in the `onAirLineRefresh()`? This is going to happen on every tick, yet `getSegmentBudgetDuration()` only depends on semi-constant `budgetDuration` on the PartInstance. Wouldn't it make more sense to calculate that re actively in the tracker, which already depends on those PartInstances, so it's going to refresh if/when they change?